pkg/gofr/logging/ctx_logger.go

```diff
@@ -0,0 +1,53 @@
+package logging
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+// ContextLogger is a wrapper around a base Logger that injects the current
+// trace ID (if present in the context) into log messages automatically.
+//
+// It is intended for use within request-scoped contexts where OpenTelemetry
+// trace information is available.
+type ContextLogger struct {
+	base Logger
+	ctx  context.Context
+}
+
+// NewContextLogger creates a new ContextLogger that wraps the provided base logger
+// and automatically appends OpenTelemetry trace information (trace ID) to log output
+// when available in the context.
+func NewContextLogger(ctx context.Context, base Logger) *ContextLogger {
+	return &ContextLogger{base: base, ctx: ctx}
+}
+
+// withTraceInfo appends the trace ID from the context (if available).
+// This allows trace IDs to be extracted later during formatting or filtering.
+func (l *ContextLogger) withTraceInfo(args ...any) []any {
+	sc := trace.SpanFromContext(l.ctx).SpanContext()
+	if sc.IsValid() {
+		return append(args, map[string]any{"__trace_id__": sc.TraceID().String()})
+	}
+
+	return args
+}
+
+func (l *ContextLogger) Debug(args ...any)            { l.base.Debug(l.withTraceInfo(args...)...) }
+func (l *ContextLogger) Debugf(f string, args ...any) { l.base.Debugf(f, l.withTraceInfo(args...)...) }
+func (l *ContextLogger) Log(args ...any)              { l.base.Log(l.withTraceInfo(args...)...) }
+func (l *ContextLogger) Logf(f string, args ...any)   { l.base.Logf(f, l.withTraceInfo(args...)...) }
```

Comment:
Here everything is repeated it doesn't help with readability and maintenance

Please consider something like


```suggestion
func (l *ContextLogger) logWithTraceID(lf func(args ...any)) {
lf(l.withTraceInfo(args...)...)
}

func (l *ContextLogger) logWithTraceIDf(lf func(f string, args ...any)) {
lf(f, l.withTraceInfo(args...)...)
}

func (l *ContextLogger) Debug(args ...any)            { l.logWithTraceID(l.base.Debug) }
func (l *ContextLogger) Debugf(f string, args ...any) { l.logWithTraceIDf(l.base.Debugf) }
func (l *ContextLogger) Log(args ...any)            { l.logWithTraceID(l.base.Log) }
func (l *ContextLogger) Logf(f string, args ...any) { l.logWithTraceIDf(l.base.Logf) }
```